runner.go

```diff
@@ -270,6 +248,32 @@ func (runner *Runner) StopWorker(id string) error {
 	return ErrDead
 }
 
+// StopAndRemoveWorker stops the worker and returns any error reported by
+// the worker, waiting for the worker to be no longer known to the runner.
+// If it was stopped while waiting, StopAndRemoveWorker will return ErrAborted.
+//
+// StopAndRemoveWorker returns ErrDead if the runner is not running.
+func (runner *Runner) StopAndRemoveWorker(id string, abort <-chan struct{}) error {
+	// workerInfo contains the done channel to signal that the worker has
+	// been removed from the runner.
+	// If it's not there, then Worker() returns not found anyway.
+	runner.mu.Lock()
+	workerInfo, ok := runner.workers[id]
+	runner.mu.Unlock()
+	w, err := runner.Worker(id, abort)
+	if err != nil || !ok {
+		return err
+	}
+	workerErr := Stop(w)
+
+	select {
+	case <-abort:
+	case <-workerInfo.done:
```

Comment:
should we handle the tomb dying here too?

Reply:
yeah, added

Reply:
actually, we don't need it here cause the workerInfo() does that. When the last worker stops the runner dies and there's a race if we try and capture the dying error here